consensus/spos/worker.go

```diff
@@ -749,6 +750,19 @@ func (wrk *Worker) processEquivalentMessage(msgType consensus.MessageType, block
 	return nil
 }
 
+func (wrk *Worker) processInvalidEquivalentMessage(msgType consensus.MessageType, blockHeaderHash []byte) {
+	if !wrk.consensusService.IsMessageWithFinalInfo(msgType) {
+		return
+	}
+
+	hdrHash := string(blockHeaderHash)
+
+	wrk.mutEquivalentMessages.Lock()
+	defer wrk.mutEquivalentMessages.Unlock()
+
+	delete(wrk.equivalentMessages, hdrHash)
```

Comment:
I do not think this is the proper way to do this.
Reason: what happens in this scenario?
1. receive the correct message
2. receive the same block final info but for some reason `checkConsensusMessageValidity` returns an error. We delete the equivalent message for the hdrHash
3. we receive the correct message again and we re-broadcast it

Reply:
pushed